### PR TITLE
feat: SQDSKDS-4877 - Rebuiding state for individual GDPR consents for OneTrust

### PR DESCRIPTION
**Pull request by markvdouw**
SQDSKDS-4877 - Rebuiding state for individual GDPR consents for OneTrust

## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 Adding changes to merge the current consent state with the individual consents selected on OneTrust.

Need to rebuild the overall consent state and re-set it to the MParticleUser, because the enable/disable actions for the individual GDPR consents are received one by one. Adding the corresponding tests.

 ## Testing Plan
 - [] Was this tested locally? If not, explain why.
 - Unit test created and checked logcat/livestream

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://mparticle-eng.atlassian.net/browse/SQDSDKS-4877

### CODE REVIEW

**Review by samdozor — CHANGES_REQUESTED** [review]
See my comment inline - thank you!

File: src/main/kotlin/com/mparticle/kits/OneTrustKit.kt [hunk 1]
```diff
@@ -254,20 +281,25 @@ class OneTrustKit : KitIntegration(), IdentityStateListener {
...
+            } else {
+                cookieValue to OneTrustConsent(purpose, regulation)
+            }
+        }
             .filterNotNull()
             .associate { it.first to it.second }
     }
+
+    override fun onUserIdentified(user: MParticleUser, previousUser: MParticleUser?) {
+
```
Comment (samdozor) [inline src/main/kotlin/com/mparticle/kits/OneTrustKit.kt:302]: I think you want a `processOneTrustConsents()` here. we store consent state at the MPID level, and so we want to make sure whenever MPID changes (and this method gets called when that happens), we are syncing the onetrust state to the user.

**Review by samdozor — CHANGES_REQUESTED** [review]
overall looks great! just one easy comment

File: src/main/kotlin/com/mparticle/kits/OneTrustKit.kt (hunk 1 above)
Comment (markvdouw) [reply src/main/kotlin/com/mparticle/kits/OneTrustKit.kt:302]: oh ok perfect ill add it

**Review by samdozor — APPROVED** [review]

**Pull request merged**